docs/compatibility_doc.md

```diff
@@ -0,0 +1,145 @@
+<!-- Generated with Stardoc: http://skydoc.bazel.build -->
+
+Skylib module of convenience functions for `target_compatible_with`.
+
+Load the macros as follows in your `BUILD` files:
+```python
+load("@bazel_skylib//lib:compatibility.bzl", "compatibility")
+```
+
+See the [Platform docs](https://bazel.build/docs/platforms#skipping-incompatible-targets) for
+more information.
+
+
+<a id="#compatibility.all_of"></a>
+
+## compatibility.all_of
```

Comment:
Is this function really necessary? 
I think the example should work with `target_compatible_with =[":dbg", "@platforms//os:linux"]`, that is by default conjunction is used.

Reply:
This supports any `config_setting`, not just constraints. I think `target_compatible_with` only works with just a list of constraints by default.